packages/grid/x-data-grid/src/internals/utils/computeSlots.ts

```diff
@@ -0,0 +1,29 @@
+import { uncapitalizeObjectKeys, UncapitalizeObjectKeys } from './slotsMigration';
+
+// TODO v7: Remove `components`, rename defaultComponents to `defaultSlots`
+// and update slots to be `slots?: <Partial<SlotComponents>` after converting
+// keys in Grid(Pro|Premium)SlotsComponent to camelCase.
+// https://github.com/mui/mui-x/issues/7940
+export function computeSlots<SlotComponents extends object>({
+  defaultComponents,
+  slots,
+  components,
+}: {
+  defaultComponents: SlotComponents;
+  slots?: UncapitalizeObjectKeys<Partial<SlotComponents>>;
+  components?: Partial<SlotComponents>;
+}): UncapitalizeObjectKeys<SlotComponents> {
+  const uncapitalizedDefaultSlots = uncapitalizeObjectKeys(defaultComponents)!;
+  const overrides = slots ?? (components ? uncapitalizeObjectKeys(components) : null);
+
+  if (!overrides) {
+    return { ...uncapitalizedDefaultSlots };
+  }
+
+  type OverrideKeys = keyof UncapitalizeObjectKeys<Partial<SlotComponents>>;
+  return Object.entries(uncapitalizedDefaultSlots).reduce((acc, [key, defaultComponent]) => {
```

Comment:
Couldn't we replace `Object.entries().reduce` with this?
```tsx
return { ...defaultSlots, ...overrides };
```

Reply:
Indeed, it seems like a simplistic thing to do 👍 
Probably previously looping over default slot keys was done as a safety check to only honor supported keys even if the users pass extra ones, but I agree removing that logic doesn't hurt much, removed!
Not sure if it still makes sense to keep the tiny `computeSlots` though. 😆

Reply:
> Not sure if it still makes sense to keep the tiny computeSlots function though. 😆

Haha, it made way more sense when it was large 😀
But it's still better to have it in one place 👍